src/scicookie/{{cookiecutter.project_slug}}/build-system/hatch-pyproject.toml

```diff
@@ -0,0 +1,128 @@
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: 3.12",
+  "Topic :: Scientific/Engineering",
+  "Typing :: Typed",
+]
+version = "{{ cookiecutter.project_version }}"
+requires-python = ">=3.8.1"
+dependencies = [
+{# keep this line here #}
+{%- if cookiecutter.use_pytest == "yes" -%}
+    "pytest >= 7.3.2",
+{% if cookiecutter.use_coverage == "yes" -%}
+    "pytest-cov >= 4.1.0",
+{% endif %}
+{%- endif -%}{#- end of use_pytest -#}
+{%- if cookiecutter.use_hypothesis == "yes" -%}
+    "hypothesis >= 6.0",
+{% endif %}
+{%- if cookiecutter.use_coverage == "yes" -%}
+    "coverage >= 7.2.7",
+{% endif %}
+{%- if cookiecutter.use_blue == "yes" -%}
+    "blue >= 0.9.1",
+{% endif %}
+{%- if cookiecutter.use_black == "yes" -%}
+    "black >= 23.3.0",
+{% endif %}
+{%- if cookiecutter.use_isort == "yes" -%}
+    "isort >= 5.12.0",
+{% endif %}
+{%- if cookiecutter.use_pre_commit -%}
+    "pre-commit >= 3.3.2",
+{% endif %}
+{%- if cookiecutter.use_flake8 == "yes" -%}
+    "flake8 >= 4.0.1, < 7",
+{% endif %}
+{%- if cookiecutter.use_ruff == "yes" -%}
+    "ruff >= 0.0.272",
+{% endif %}
+{%- if cookiecutter.use_mypy == "yes" -%}
+    "mypy >= 1.3.0",
+{% endif %}
+{%- if cookiecutter.use_bandit == "yes" -%}
+    "bandit >= 1.7.5",
+{% endif %}
+{%- if cookiecutter.use_pydocstyle == "yes" -%}
+    "pydocstyle >= 6.3.0",
+{% endif %}
+{%- if cookiecutter.use_vulture == "yes" -%}
+    "vulture >= 2.7",
+{% endif %}
+{%- if cookiecutter.use_mccabe == "yes" -%}
+    "mccabe >= 0.6.1",
+{% endif %}
+{%- if cookiecutter.use_containers in ['Docker', 'Podman'] -%}
+# if you want to use docker-compose from your system, remove compose-go here
+    "compose-go >= 2.18.1",
+{% endif %}
+{%- if cookiecutter.documentation_engine == 'mkdocs' -%}
+    "Jinja2 >= 3.1.2",
+    "mkdocs >= 1.4.3",
+    "mkdocs-exclude >= 1.0.2",
+    "mkdocs-jupyter >= 0.24.1",
+    "mkdocs-literate-nav >= 0.6.0",
+    "mkdocs-macros-plugin >= 0.7.0, < 1",
+    "mkdocs-material >= 9.1.15",
+    "mkdocstrings >= 0.21.2",
+    "mkdocstrings-python >= 1.1.2",
+{% elif cookiecutter.documentation_engine == 'sphinx' -%}
+    "Sphinx >= 6.2.1",
+    "sphinx-rtd-theme >= 1.2.2",
+    "importlib-metadata >= 6.5.1",
+    "myst-parser >= 0.19.2",
+    "nbsphinx >= 0.9.2",
+    "pandoc >= 2.3",
+{% elif cookiecutter.documentation_engine == 'jupyter-book' -%}
+    "jupyter-book >= 0.15.1",
+    "myst-parser >= 0.18.1",
+{% endif %}
+]
+
+[project.urls]
+Homepage = "{{ cookiecutter.project_url }}"
+"Bug Tracker" = "{{ cookiecutter.project_url }}/issues"
+Discussions = "{{ cookiecutter.project_url }}/discussions"
+Changelog = "{{ cookiecutter.project_url }}/releases"
+
+[tool.hatch.version]
+{# keep this line here #}
+{%- if cookiecutter.project_layout == 'flat' -%}
+path = "{{ cookiecutter.package_slug }}/__init__.py"
+{% elif cookiecutter.project_layout == 'src' -%}
+path = "src/{{ cookiecutter.package_slug }}/__init__.py"
+{% endif %}
```

Comment:
This won't be required if we end up using static versioning for Hatch.

Reply:
@Saransh-cpp could you add more information here please?
I am not following you, sorry

Reply:
Sorry for the confusion. These lines are only required when we use dynamic version in `pyproject.toml` -
```toml
dynamic = ["version"]
```
but as this PR now uses static version -

https://github.com/osl-incubator/scicookie/blob/32704a5fba35869f5a67c6b7db76f933e9abd211/src/scicookie/%7B%7Bcookiecutter.project_slug%7D%7D/build-system/hatch-pyproject.toml#L37

the whole `[tool.hatch.version]` section is basically redundant.